bindings/python/scripts/utils.py

```diff
@@ -37,6 +37,11 @@ def get_output_path(source, output_dir, split_from, extension):
     # split_path: contains the path after splitting. For split_path = pcl, contains the path as seen in the pcl directory
     _, split_path = source.split(f"{split_from}{os.sep}", 1)
 
+    # hack for CI; it contains multiple `pcl`s:
+    # `/home/runner/work/pcl/pcl/bindings/python` instead of `/home/runner/work/pcl/bindings/python`
+    if split_path.startswith("pcl"):
```

Comment:
Is something weird? In generate.py, split_from is hardcoded as json

Reply:
For parse.py, passing point_types.cpp:
```
CI generates this path: /home/runner/work/pcl/pcl/bindings/python/pybind11-gen/common/include/pcl/impl/point_types.cpp
Local build generates this path: /home/divyanshu/Projects/active/pcl/bindings/python/pybind11-gen/common/include/pcl/impl/point_types.cpp
```
Split from `pcl`:
```
CI: pcl/bindings/python/pybind11-gen/common/include/pcl/impl/point_types.cpp
Local: bindings/python/pybind11-gen/common/include/pcl/impl/point_types.cpp
```
An extra `pcl`, remove.

***

After parsing, to generate.py:
```
CI generates this path: /home/runner/work/pcl/pcl/bindings/python/json/pcl/common/include/pcl/impl/point_types.json
Local build generates this path: /home/divyanshu/Projects/active/pcl/bindings/python/json/common/include/pcl/impl/point_types.json
```
Split from `json`:
```
CI: pcl/common/include/pcl/impl/point_types.json
Local: common/include/pcl/impl/point_types.json
```
An extra `pcl`, remove.

Reply:
This issue would be removed if you just take the root directory as optional apart from the module name. If root_directory is provided, split from that, else split from module name